zproject/urls.py

```diff
@@ -432,146 +671,210 @@
-            zerver.views.unsubscribe.email_unsubscribe,
-            name='zerver.views.unsubscribe.email_unsubscribe'),
-
+    re_path(
+        r'^accounts/unsubscribe/(?P<email_type>[\w]+)/(?P<confirmation_key>[\w]+)$',
+        zerver.views.unsubscribe.email_unsubscribe,
+        name='zerver.views.unsubscribe.email_unsubscribe',
+    ),
     # Portico-styled page used to provide email confirmation of terms acceptance.
-    path('accounts/accept_terms/', zerver.views.home.accounts_accept_terms,
-         name='zerver.views.home.accounts_accept_terms'),
-
+    path(
+        'accounts/accept_terms/',
+        zerver.views.home.accounts_accept_terms,
+        name='zerver.views.home.accounts_accept_terms',
+    ),
     # Find your account
-    path('accounts/find/', zerver.views.registration.find_account,
-         name='zerver.views.registration.find_account'),
-
+    path(
+        'accounts/find/',
+        zerver.views.registration.find_account,
+        name='zerver.views.registration.find_account',
+    ),
     # Go to organization subdomain
-    path('accounts/go/', zerver.views.registration.realm_redirect,
-         name='zerver.views.registration.realm_redirect'),
-
+    path(
+        'accounts/go/',
+        zerver.views.registration.realm_redirect,
+        name='zerver.views.registration.realm_redirect',
+    ),
     # Realm Creation
-    path('new/', zerver.views.registration.create_realm,
-         name='zerver.views.create_realm'),
-    re_path(r'^new/(?P<creation_key>[\w]+)$',
-            zerver.views.registration.create_realm, name='zerver.views.create_realm'),
-
+    path('new/', zerver.views.registration.create_realm, name='zerver.views.create_realm'),
+    re_path(
+        r'^new/(?P<creation_key>[\w]+)$',
+        zerver.views.registration.create_realm,
+        name='zerver.views.create_realm',
+    ),
     # Realm Reactivation
-    re_path(r'^reactivate/(?P<confirmation_key>[\w]+)$', zerver.views.realm.realm_reactivation,
-            name='zerver.views.realm.realm_reactivation'),
-
+    re_path(
+        r'^reactivate/(?P<confirmation_key>[\w]+)$',
+        zerver.views.realm.realm_reactivation,
+        name='zerver.views.realm.realm_reactivation',
+    ),
     # Global public streams (Zulip's way of doing archives)
-    path('archive/streams/<int:stream_id>/topics/<str:topic_name>',
-         zerver.views.archive.archive,
-         name='zerver.views.archive.archive'),
-    path('archive/streams/<int:stream_id>/topics',
-         zerver.views.archive.get_web_public_topics_backend,
-         name='zerver.views.archive.get_web_public_topics_backend'),
-
+    path(
+        'archive/streams/<int:stream_id>/topics/<str:topic_name>',
+        zerver.views.archive.archive,
+        name='zerver.views.archive.archive',
+    ),
+    path(
+        'archive/streams/<int:stream_id>/topics',
+        zerver.views.archive.get_web_public_topics_backend,
+        name='zerver.views.archive.get_web_public_topics_backend',
+    ),
```

Comment:
We may want to think about excluding this file.  Though on thing on my mind is whether these `name` parameters are useless?  If so, we should probably delete those before looking at what Black does.

Reply:
`name` only exists for `reverse()`, so I’m guessing most of these were cargo-culted. We don’t call `reverse('zerver.views.archive.get_web_public_topics_backend')`. If we did, it should probably instead be `reverse(zerver.views.archive.get_web_public_topics_backend)` which doesn’t need a `name`. Figuring out which `name`s we do use will take some auditing.

Reply:
Yeah, we hardly use `reverse(`, which is why I believe most of these are useless.  I think it's basically only used in the login flow code path.

Reply:
I’ve submitted some cleanup ideas as #16389.